consensus/polybft/system_state/system_state.go

```diff
@@ -107,38 +107,166 @@ func (s *SystemStateImpl) GetNextCommittedIndex(chainID uint64, chainType ChainT
 func (s *SystemStateImpl) GetBridgeBatchByNumber(numberOfBatch *big.Int) (
 	*contractsapi.SignedBridgeMessageBatch, error) {
 	rawResult, err := s.bridgeStorageContract.Call(
-		"batches",
+		"getCommittedBatch",
 		ethgo.Latest,
 		numberOfBatch)
 	if err != nil {
 		return nil, err
 	}
 
-	bridgeBatch, isOk := rawResult["0"].(*contractsapi.SignedBridgeMessageBatch)
-	if !isOk {
-		return nil, fmt.Errorf("failed to decode bridge batch")
+	decErr := fmt.Errorf("failed to decode")
+
+	rawResult, ok := rawResult["0"].(map[string]interface{})
```

Comment:
Didn't we agree that we will add a new function to ethgo that will receive an interface that has a DecodeAbi function that receives a byte array, so that we define how a call is decoded? I prefer that approach, since we use our own fork of ethgo.

Reply:
@grujicf this is just a draft idea and it needs polishing (https://github.com/Ethernal-Tech/blade/pull/401/commits/61b5ff415b008172ee79500bd5ab9cb2a5dbc6c2) as the CI might be failing.

Also, please take a look to this PR and figure out what is wrong with the tests (https://github.com/Ethernal-Tech/ethgo/pull/18).